File: backend/src/main/java/me/performancereservation/domain/review/repository/ReviewRepository.java
package me.performancereservation.domain.review.repository;

import me.performancereservation.domain.review.Review;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface ReviewRepository extends JpaRepository<Review, Long> {
    //리뷰 페이징
    Page<Review> findByPerformanceIdOrderByCreatedAtDesc(Long performanceId, Pageable pageable);

    boolean existsByUserIdAndPerformanceId(Long userId, Long performanceId);
}

File: backend/src/main/java/me/performancereservation/domain/review/repository/ReviewRepository.kt
package me.performancereservation.domain.review.repository

import me.performancereservation.domain.review.Review
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface ReviewRepository : JpaRepository<Review, Long> {
    fun findByPerformanceIdOrderByCreatedAtDesc(performanceId: Long?, pageable: Pageable?): Page<Review>

    fun existsByUserIdAndPerformanceId(userId: Long, performanceId: Long): Boolean
}
